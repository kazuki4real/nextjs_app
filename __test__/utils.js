import { render as defaultRender } from '@testing-library/react'
import { RouterContext } from 'next/dist/next-server/lib/router-context'
import React from 'react'

export function render(children) {
  const mockRouter = {
    route: '/',
    pathname: '/',
    query: {},
    asPath: '/',
    basePath: '/',
    isLocaleDomain: true,
    isReady: true,
    push: jest.fn(),
    prefetch: jest.fn(),
    replace: jest.fn(),
    reload: jest.fn(),
    back: jest.fn(),
    beforePopState: jest.fn(),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
    isPreview: false,
  }

  return defaultRender(
    <RouterContext.Provider value={mockRouter}>
      {children}
    </RouterContext.Provider>,
  )
}
