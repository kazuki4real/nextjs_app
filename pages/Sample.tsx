import React from 'react'
import { useRouter } from 'next/router'

const Sample = () => {
  const router = useRouter()
  const path = router.asPath
  return <div>このページのパスは{path}です。</div>
}

export default Sample
