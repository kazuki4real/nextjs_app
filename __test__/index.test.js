import Sample from '../pages/Sample'
import { render } from './utils'

describe('初期表示の確認', () => {
  test('サンプルページ', () => {
    const { asFragment } = render(<Sample />)
    expect(asFragment()).toMatchSnapshot()
  })
})
