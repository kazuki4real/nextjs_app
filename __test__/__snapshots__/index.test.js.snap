// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`初期表示の確認 サンプルページ 1`] = `
<DocumentFragment>
  <div>
    このページのパスは/です。
  </div>
</DocumentFragment>
`;
